six/modules/c++/samples/test_create_sidd_from_mem.cpp

```diff
@@ -1773,17 +1774,498 @@ void initProcessorInformation(
+    //
+    //     | y^0  y^1 ... y^n
+    //-------------------------------
+    // x^0 |  7    0   0   0
+    // x^1 |  0    3   0   0
+    // ... |  0    0   0   0
+    // x^n |  2    0   0   0
+    //--------------------------------------------------
+    planeProjection->timeCOAPoly = six::Poly2D(0, 0);
+    planeProjection->timeCOAPoly[0][0] = 1;
+
+    //--------------------------------------------------
+    // This is creating a constant-term polynomial of vectors
+    // In this case, an index into the polynomial yields a
+    // six::Vector3
+    //--------------------------------------------------
+    siddData.measurement->arpPoly = six::PolyXYZ(0);
+
+    // The constant term is a vector.  Each component is 0
+    siddData.measurement->arpPoly[0] = six::Vector3(0.0);
+
+    //----------------------------------------------------
+    // The basis vectors are dummied, each component is 0
+    // This works because Vector3 supports assignment from
+    // a scalar, assuming that each component should be
+    // set to that value.  This is not really math, just
+    // C++ syntactic sugar.  Vector3 supports most math
+    // operations that one would expect
+    //----------------------------------------------------
+    planeProjection->productPlane.rowUnitVector = six::Vector3(0.0);
+    planeProjection->productPlane.colUnitVector = six::Vector3(0.0);
+
+    initExploitationFeatures(*siddData.exploitationFeatures);
+    initProductProcessing(*siddData.productProcessing);
+    initDownstreamReprocessing(*siddData.downstreamReprocessing);
+    initErrorStatistics(*siddData.errorStatistics);
+    initRadiometric(*siddData.radiometric);
+    initAnnotations(siddData.annotations);
+}
+
+int main(int argc, char** argv)
+{
+    cli::ArgumentParser argParser;
+    argParser.setDescription("Generate a NITF from specified options");
+    argParser.addArgument("--lut", "Specify no lut, mono lut, or color lut",
+            cli::STORE, "lut", "None, Mono, or Color", 0, 1)->
+            setChoices(str::split("None Mono Color"))->setDefault("None");
+    argParser.addArgument("--multipleSegments",
+            "Single image, multiple segments", cli::STORE_TRUE,
+            "multipleSegments", "", 0, 1);
+    argParser.addArgument("--multipleImages",
+            "Add multiple images to NITF", cli::STORE_TRUE,
+            "multipleImages", "", 0, 1);
+    argParser.addArgument("--smallImage", "Use 2x2 image with dummy data",
+            cli::STORE_TRUE, "smallImage", "", 0, 1);
+    argParser.addArgument("output", "File to write to", cli::STORE, "output",
+            "output-file", 1, 1, true);
+    argParser.addArgument("xml", "Optional SICD .xml file", cli::STORE,
+            "sicdXML", "sicd-xml", 0, 1);
+
+    //We'll get this from command line args once it matters
+    const std::string version("1.0.0");
 
     try
     {
+        std::auto_ptr<cli::Results> options(argParser.parse(argc,
+                (const char**) argv));
```

Comment:
Don't need the `const char**` cast here (old code might have been doing this b/c we used to not have an overloading)